File: examples/E07-sobrecarga-e-parametros-default/expected/lib/formatacao.dart
String formatarValorInt(int valor) {
  return 'inteiro';
}

String formatarValorDouble(double valor) {
  return 'real';
}

int incrementar(int valor, [int passo = 1]) {
  return valor + passo;
}

String testarFormatoInt() {
  return formatarValorInt(5);
}

String testarFormatoDouble() {
  return formatarValorDouble(5);
}

int testarIncrementoPadrao() {
  return incrementar(10);
}

int testarIncrementoComPasso() {
  return incrementar(10, 5);
}

File: examples/E07-sobrecarga-e-parametros-default/input/src/formatacao.cpp
#include "formatacao.hpp"

std::string formatarValor(int valor) {
    return "inteiro";
}

std::string formatarValor(double valor) {
    return "real";
}

int incrementar(int valor, int passo) {
    return valor + passo;
}

std::string testarFormatoInt() {
    return formatarValor(5);
}

std::string testarFormatoDouble() {
    return formatarValor(5.0);
}

int testarIncrementoPadrao() {
    return incrementar(10);
}

int testarIncrementoComPasso() {
    return incrementar(10, 5);
}
